src/helix-ui/elements/HXTooltipElement.js

```diff
@@ -0,0 +1,162 @@
+    }
+
+    disconnectedCallback () {
+        if (!this._target) {
+            return;
+        }
+        this._destoryAllHandlers();
+    }
+
+    attributeChangedCallback (attr, oldValue, newValue) {
+        this.setAttribute('aria-hidden', newValue !== '');
+    }
+
+    _hide () {
+        if (this._showTimer) {
+            clearTimeout(this._showTimer);
+        }
+        this._hideTimer = setTimeout(() => {
+            this.open = false;
+        }, 1600);
+    }
+
+    _show () {
+        if (this._hideTimer) {
+            clearTimeout(this._hideTimer);
+        }
+        this._showTimer = setTimeout(() => {
+            this.open = true;
+        }, 500);
+    }
+
+    _toggle () {
+        this.open = !this.open;
+    }
+
+    _closeOnBackgroundClick (event) {
+        if (this._isBackground(event)) {
+            this.open = false;
+        }
+    }
+
+    _connectHandlers (type) {
+        window.addEventListener('resize', debounce(this._setPosition,100));
+        if (type === 'click') {
+            document.addEventListener('click', this._closeOnBackgroundClick);
+            this._target.addEventListener('click', this._toggle);
+        } else {
+            this._target.addEventListener('focus', this._show);
+            this._target.addEventListener('blur', this._hide);
+            this._target.addEventListener('mouseenter', this._show);
+            this._target.addEventListener('mouseleave', this._hide);
+        }
+    }
+
+    _destoryAllHandlers () {
+        window.removeEventListener('resize', debounce(this._setPosition,100));
+        document.removeEventListener('click', this._closeOnBackgroundClick);
+        this._target.removeEventListener('focus', this._show);
+        this._target.removeEventListener('blur', this._hide);
+        this._target.removeEventListener('mouseenter', this._show);
+        this._target.removeEventListener('mouseleave', this._hide);
+        this._target.removeEventListener('click', this._toggle);
+    }
+
+    _setPosition () {
+        var offset = getPositionWithArrow(this, this._target, { 'position':this.position });
+        this.style.top = `${offset.y}px`;
+        this.style.left = `${offset.x}px`;
+        this.position = offset.position;
+    }
+
+    _isBackground (event) {
+        let inComponent = this.contains(event.target);
+        let inTarget = this._target.contains(event.target);
+        return !inComponent && !inTarget ;
+    }
+
+    set position (value) {
+        if (value) {
+            this.setAttribute('position', value);
+        } else {
+            this.removeAttribute('position');
+        }
+    }
+
+    get position () {
+        return this.getAttribute('position');
+    }
+
+    set trigger (value) {
+        if (value) {
+            this.setAttribute('trigger', value);
+        } else {
+            this.removeAttribute('trigger');
+        }
+    }
+
+    get trigger () {
+        return this.getAttribute('trigger');
```

Comment:
Maybe we should rename the attribute to `trigger-event` (with a `triggerEvent` property) to make it more explicit what the attribute is configuring.